### PR TITLE
Make use of capitano's promise support to simplify the code

### PR DESCRIPTION
**Pull request by Page-**
There is still usage of the callback method for capitano in the codebase but only for code that actually uses a callback interface somewhere, anything purely promise based now relies on the promise handling (and as you can see I got to delete a load of code :D )

### CODE REVIEW

File: lib/actions/api-key.ts [hunk 1]
```diff
@@ -18,8 +18,8 @@ export const generate: CommandDefinition<{
 
 			$ balena api-key generate "Jenkins Key"
 	`,
-	async action(params, _options, done) {
-		getBalenaSdk()
+	async action(params) {
```
Comment (pdcastro) [inline lib/actions/api-key.ts]: I haven't tested this to confirm so I don't know for sure that it doesn't work, but... I see that Capitano detects promises by testing whether the "promise object" has a `.then` property:
```
					if callback?
						if actionPromise?.then?
							# Asynchronous action function
							actionPromise.then(
```
And while this test works with Bluebird promises, it doesn't seem to work with native async promises / functions. So perhaps Capitano would need to be patched for some of the changes in this PR to work. Needs confirmation.

Comment (pdcastro) [reply lib/actions/api-key.ts]: Testing with the interactive interpreter, note how **`f.then`** is `undefined`, whereas **`g.then`** is `Function`:
```js
$ node
> async function f() {
... await new Promise(resolve => setTimeout(resolve, 1000));
... }
undefined
> f.then
undefined
> const Bluebird = require('bluebird');
undefined
> const g = Bluebird.resolve(new Promise(resolve => setTimeout(resolve, 1000)));
Promise [Object] {
  _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined }
> g.then
[Function]
```

Comment (pdcastro) [reply lib/actions/api-key.ts]: Perhaps the Capitano fix would be as simple as replacing:
```coffeescript
if actionPromise?.then?
```
with:
```coffeescript
if actionPromise?.then? or actionPromise[Symbol.toStringTag] == 'AsyncFunction'
```
on this line: https://github.com/balena-io/capitano/blob/v1.9.1/lib/command.coffee#L78

Comment (srlowe) [reply lib/actions/api-key.ts]: Yeah I think checking for `.then()` is the most common way to detect a promise object cross-implementation.

Comment (srlowe) [reply lib/actions/api-key.ts]: I thought testing for presence of `.then()` was the most common implementation-agnostic way of detecting promise object.

Comment (srlowe) [reply lib/actions/api-key.ts]: @pdcastro In your above example you are testing `f.then`, but shouldn't you be testing `f().then`?

e.g. this line in capitano: https://github.com/balena-io/capitano/blob/8547387f6b62c8a5777da07f9e29236568ebe57b/lib/command.coffee#L73

Comment (Page-) [reply lib/actions/api-key.ts]: @pdcastro as @srlowe points out your test is flawed, you're testing for `Function.then` not `Promise.then`. The code definitely works, eg check out https://github.com/balena-io/balena-cli/blob/v11.28.5/lib/actions/auth.ts#L85 which is already using async with capitano's promise support

Comment (pdcastro) [reply lib/actions/api-key.ts]: @srlowe, yes, you're right, I should have tested `f().then`, in which case the test works! Thanks! Now approving the PR. :-)

**Pull request merged**